src/server/race/RaceDirector.cpp

```diff
@@ -1331,6 +1335,82 @@ void RaceDirector::HandleStartRace(
   if (clientContext.characterUid != raceInstance.masterUid)
     throw std::runtime_error("Client tried to start the race even though they're not the master");
 
+  // Check if all requirements are met to start the race
+  bool allPlayersReady = false;
+  bool isTeamMode = false;
+  uint32_t redTeamCount = 0;
+  uint32_t blueTeamCount = 0;
+
+  _serverInstance.GetRoomSystem().GetRoom(
+    clientContext.roomUid,
+    [&allPlayersReady, &isTeamMode, &redTeamCount, &blueTeamCount, clientContext](Room& room)
```

Comment:
```suggestion
    [&allPlayersReady, &isTeamMode, &redTeamCount, &blueTeamCount, &clientContext](Room& room)
```
I think we should pass this as a reference.